Mathlib/Data/IsROrC/Lemmas.lean

```diff
@@ -38,14 +38,8 @@ This instance generates a type-class problem with a metavariable `?m` that shoul
 /-- An `IsROrC` field is finite-dimensional over `ℝ`, since it is spanned by `{1, I}`. -/
 -- Porting note: was @[nolint dangerous_instance]
 instance isROrC_to_real : FiniteDimensional ℝ K :=
-  ⟨⟨{1, I}, by
-      rw [eq_top_iff]
-      intro a _
-      rw [Finset.coe_insert, Finset.coe_singleton, Submodule.mem_span_insert]
-      refine' ⟨re a, im a • I, _, _⟩
-      · rw [Submodule.mem_span_singleton]
-        use im a
-      simp [re_add_im a, Algebra.smul_def, algebraMap_eq_ofReal]⟩⟩
+  ⟨{1, I}, by simpa [Submodule.eq_top_iff', Submodule.mem_span_pair] using
+    fun x ↦ ⟨re x, im x, by simp [real_smul_eq_coe_mul]⟩⟩
```

Comment:
This proof relies a lot on the fact that `simpa` doesn't simplify the expression too much. I would prefer `suffices _ by simpa [...]; exact _` or `simp? [] says ...; exact`

Reply:
@urkud I've changed it to use `suffices`, but for what it's worth, I disagree with:

> This proof relies a lot on the fact that `simpa` doesn't simplify the expression too much.

The suffices I added was `∀ x : K, ∃ a b : ℝ, a • 1 + b • I = x` (the goal after `simp`) which could only really be simplified further if we made `IsROrC.real_smul_eq_coe_mul` a `simp` lemma. But even if that *did* happen, this would be covered by the `by simp [real_smul_eq_coe_mul]`. Note also, making that a `simp` lemma would be a bit weird since it's *inserting* a coercion.